File: updater/src/main/java/updater/identifiers/MusicPatchNode2.java
package updater.identifiers;

import org.objectweb.asm.Type;
import updater.utility.AbstractIdentifier;
import updater.utility.ClassWrapper;
import updater.utility.TypeUtilities;

public class MusicPatchNode2 extends AbstractIdentifier {
    @Override
    public boolean isMatch(ClassWrapper classNode) {
        return classNode.isOwnerless() &&
                classNode.getCountFieldsOfType(TypeUtilities.BYTE_ARRAY) == 2 &&
                classNode.getCountFieldsOfType(Type.INT_TYPE) == 7;
    }
}

File: updater-runestar/src/main/java/org/runestar/client/updater/mapper/identifiers/disabled/MusicPatchNode2.kt
package org.runestar.client.updater.mapper.identifiers.disabled

import org.objectweb.asm.Type
import org.runestar.client.updater.mapper.abstractclasses.IdentityMapper
import org.runestar.client.updater.mapper.predicateutilities.and
import org.runestar.client.updater.mapper.predicateutilities.predicateOf
import org.runestar.client.updater.mapper.predicateutilities.type
import org.runestar.client.updater.mapper.wrappers.Class2

class MusicPatchNode2 : IdentityMapper.Class() {

    override val predicate = predicateOf<Class2> { it.superType == Any::class.type }
            .and { it.instanceFields.count { it.type == ByteArray::class.type } == 2 }
            .and { it.instanceFields.count { it.type == Type.INT_TYPE } == 7 }
}